### PR TITLE
[81X] Add Phase-I Cosmics

### PR DESCRIPTION
**Pull request by mmusich**
This PR attempts to add a new cosmics simulation workflow using phase-I 2017 detector.
In order to consume cosmics reconstruction-tailored conditions a new autoCond Global Tag is added. 
To address comments occurring during the review, appropriate commits were included to handle Hcal 2017 reconstruction with cosmics data and propagate to this wf the new phase-I pixel DQM.  
On Hcal side (thanks @kpedro88):
   * Increased flexibility of HCAL cosmic setup by applying cosmic-specific settings to cloned modules and then assembling sequences for 2017 and pre-2017 separately
   * Updated HCAL NZS sequence for 2017 HF and HE changes (can be used in cosmic runs)
   * Modified size of simulated QIE10DigiCollection to be 4 samples instead of default 10, by accessing sim parameter `readoutFrameSize` and forwarding through ZS algos accordingly

On DQM side (thanks @schneiml):
   * Add a special cosmics sequence to the PixelPhase1 DQM config
   * Enable it for the phase1pixel era in the general Pixel offline DQM config
   * Add a special cosmics sequence for harvesting to the Pixel offline DQM
config, since it is needed for Phase1.

### CODE REVIEW

**Comment by mmusich** [comment]
please test

**Comment by boudoul** [comment]
fantastic, thank you @kpedro88 @mmusich !

File: Configuration/PyReleaseValidation/python/relval_standard.py [hunk 1]
```diff
@@ -278,6 +278,9 @@
 
 workflows[7]  = ['', ['Cosmics','DIGICOS','RECOCOS','ALCACOS','HARVESTCOS']]
 workflows[7.1]= ['', ['CosmicsSPLoose','DIGICOS','RECOCOS','ALCACOS','HARVESTCOS']]
+workflows[7.2] = ['', ['Cosmics_UP17','DIGICOS_UP17','RECOCOS_UP17','ALCACOS_UP17','HARVESTCOS_UP17']]
+workflows[7.3] = ['', ['CosmicsSPLoose_UP17','DIGICOS_UP17','RECOCOS_UP17','ALCACOS_UP17','HARVESTCOS_UP17']]
```
Comment (slava77) [inline Configuration/PyReleaseValidation/python/relval_standard.py]: would it make sense to use the same numbering scheme as used for phase-1 MC? (1XXXX)

**Comment by kpedro88** [comment]
attn: @igv4321

**Comment by kpedro88** [comment]
@mmusich, please accept https://github.com/cms-AlCaDB/cmssw/pull/1 if you don't mind (we noticed that the NZS workflow needed to be updated for HCAL 2017, and this is also used in cosmics)

**Comment by kpedro88** [comment]
also, now there's a merge conflict with autoCond.py...

**Comment by mmusich** [comment]
@kpedro88 
rebased, though might not be the end of the rebases (other PR are competing with this one)

**Comment by mmusich** [comment]
please test

**Comment by mmusich** [comment]
@cmsbuild please test

**Comment by fabozzi** [comment]
+1

**Comment by cvuosalo** [comment]
@mmusich, @boudoul: Are revisions to this PR complete? Or can we expect more changes?

**Comment by mmusich** [comment]
@cvuosalo an update of the Global Tag is underway

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by kpedro88** [comment]
@mmusich this needs to be rebased when the next IB is out

**Comment by cvuosalo** [comment]
In step 5 of the new 7.3 workflow (step5_HARVESTING.py) there are thousands of the following messages, which seem anomalous.
```
%MSG-e HistogramManager:   SiPixelPhase1DigisHarvester:SiPixelPhase1DigisHarvester@endJob  28-Oct-2016 20:32:45 CEST PostEndRun
ME PixelPhase1/PXBarrel/adc_per_Lumisection not found

%MSG-e HistogramManager:   SiPixelPhase1TrackEfficiencyHarvester:SiPixelPhase1TrackEfficiencyHarvester@endJob  28-Oct-2016 20:32:57 CEST PostEndRun
ME PixelPhase1/PXForward/HalfCylinder_mI/PXDisk_-3/PXRing_2/PXBlade_50/clusterprob not found
```
Note, for the second error, the disk, ring, and blade numbers vary from message to message.

What is causing these errors?

**Comment by mmusich** [comment]
please test

**Comment by mmusich** [comment]
@boudoul please clarify what we should expect out of the phase-I pixel DQM for cosmics (addressing https://github.com/cms-sw/cmssw/pull/16343#issuecomment-257418584)

**Comment by boudoul** [comment]
Right I'm looking at this , calling help from @schneiml

**Comment by schneiml** [comment]
@boudoul @mmusich As far as I see there is no special config file for cosmics. I think `DQM/SiPixelCommon/python/SiPixelOfflineDQM_source_cff.py` has to be slightly adapted (see comments at the bottom there) and a selection of phase1 DQM plugins for cosmics (subset of normal offline or all?) has to be made, then we should start to get something out.

**Comment by schneiml** [comment]
@cvuosalo this is the Phase1 DQM harvesting complaining that the Phase1 DQM source does not run.

This is indeed a problem in the current/shared Pixel DQM config that there is a specailized path for Cosmics in the source but not in the Harvesting, and the Phase1 DQM can and will do arbitrary nonsense when the source and harvesting configs do not match.

However it seems to me the special cosmics config is basically identical to the default offline. So we could just get rid of it and things might work. 

I just tried it, but things fail due to the GT: 
```
Global Tag "81X_upgrade2017cosmics_realistic_peak_v4" has not been found in the database. from GTProxy::load
```

**Comment by mmusich** [comment]
@schneiml the GT is there:
https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/81X_upgrade2017cosmics_realistic_peak_v4

please try again

**Comment by schneiml** [comment]
@mmusich does indeed run now, but the only thing I can get from the DQM is a few digis. Might be incorrect collection names or even the expected behaviour.

I'll investigate what is going on.

**Comment by cvuosalo** [comment]
To test whether the new 7.3 workflow is working properly, I tried comparing it with 1307.0_CosmicsSPLoose_UP15, which is a Run 2 cosmics workflow. I ran 1000 events of each using CMSSW_8_1_X_2016-10-28-1100. Some large differences, which might indicate a problem with 7.3, are shown below. "New" is 7.3, and "Ref" is 1307.0.

all_testpr16343vsorig_cosmicsup15wf1307p0c_calotowerssorted_towermaker__reco_obj_obj_eta123

all_testpr16343vsorig_cosmicsup15wf1307p0c_hfrechitssorted_hfreco__reco_obj_obj_energy
all_testpr16343vsorig_cosmicsup15wf1307p0c_log10horechitssorted_horeco__reco_obj_obj_energy3

all_testpr16343vsorig_cosmicsup15wf1307p0c_log10hbherechitssorted_hbhereco__reco_obj_obj_energy1

**Comment by kpedro88** [comment]
@igv4321 can you check the settings I added for Phase1 HF? We don't seem to be getting any RecHits.

It's also quite odd that we have 4X more hits in HO - there shouldn't be any change in HO. Comparison of the cosmic and realistic 2017 GTs (https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/81X_upgrade2017_realistic_v21/81X_upgrade2017cosmics_realistic_peak_v4) shows they have the same HCAL conditions. I'll try to investigate this further.

**Comment by igv4321** [comment]
Hi Kevin,

Which workflow you are using? It is possible that
the rechits are simply dropped on output.

Igor

**Comment by kpedro88** [comment]
@igv4321 please refer to @cvuosalo's post above mine: he is comparing 2016 cosmic to 2017 cosmic, and we see no HFRecHits in the latter case. The InputTag hasn't changed, so I don't know why the collection would be dropped.

**Comment by igv4321** [comment]
The problem appears to be with the QIE10DataFrame samples 4 through 9.
HFSimpleTimeCheck encounters "hardware errors" there (i.e., the sample "ok()"
method returns "false") and discards the rechit. Are those time slices actually
filled out with something meaningful?

**Comment by kpedro88** [comment]
@igv4321 aha, I see. I didn't realize the HFSimpleTimeCheck had that behavior. No, those samples aren't filled; `MAX_SAMPLES = 10` is just the standard definition in `HcalDataFrameContainer` used for both QIE10 and QIE11.

Can you inform HFSimpleTimeCheck that HF only takes 4 time slices? In the simulation, there's a parameter `readoutFrameSize`, but I don't know if the corresponding information is exposed in reco.

**Comment by igv4321** [comment]
Hm... Shouldn't QIE10DataFrame be fixed instead? The number
of samples is propagated to reco from there, via the "adc2fc"
method of HcalCoderDb:

http://cmslxr.fnal.gov/source/CalibFormats/HcalObjects/src/HcalCoderDb.cc#0024

**Comment by kpedro88** [comment]
As I said, `MAX_SAMPLES = 10` is a property of the class template `HcalDataFrameContainer`, which is used for both QIE10 and QIE11 collections: https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/DataFormats/HcalDigi/interface/HcalDigiCollections.h#L41

Theoretically MAX_SAMPLES could be set as an additional template parameter, but I think a change in this data format is inadvisable right before the release closes. (In addition, ZDC/RPD is now starting to use QIE10s, and will be taking 10 samples rather than 4...)

It would be easier if we could specify the HF readout frame size as a python parameter for the reco algorithm.

**Comment by igv4321** [comment]
Hm... HcalDataFrameContainer constructor actually has a parameter specifying
the number of samples. Can the actual value be used as the constructor argument
instead of the default MAXSAMPLES value?

**Comment by kpedro88** [comment]
Maybe. It would have to be done consistently in all of these areas (digi, ZS, unpacker): https://github.com/cms-sw/cmssw/search?utf8=%E2%9C%93&q=%22new+QIE10DigiCollection%22

In `HcalDigitizer`, we can use the `readoutFrameSize` parameter. The ZS algos can infer the number of samples from the input collection, and `HcalUnpacker` infers the number of samples automatically. The only problem is making a consistent dummy collection in `HcalRawToDigi` for the case where `HcalUnpacker` doesn't find any QIE10 digis to unpack. I guess we could add a `hfReadoutFrameSize` parameter in `HcalRawToDigi`...

**Comment by kpedro88** [comment]
@jmmans do you think that is a viable plan?

**Comment by kpedro88** [comment]
I guess if it's an empty collection, the number of samples doesn't really matter.

**Comment by kpedro88** [comment]
@mmusich @igv4321 @cvuosalo I made the appropriate changes at sim level, see https://github.com/cms-AlCaDB/cmssw/pull/2. With these changes, I see a reasonable HF RecHit energy spectrum from 1000 events of WF 7.3.

**Comment by mmusich** [comment]
@cmsbuild, please test
the last updated after rebase is to pick the latest geometry changes in pixel Tracker (as in #16429)

**Comment by schneiml** [comment]
I added some patches to get Phase1 DQM output for cosmics. It is the last commit here:
https://github.com/cms-sw/cmssw/compare/CMSSW_8_1_X...schneiml:phase1pixeldqm-cosmics

**Comment by mmusich** [comment]
@schneiml thanks. I've added your commit here.

**Comment by mmusich** [comment]
please test

**Comment by smuzaffar** [comment]
Looks like comparison for this failed to upload the results. I have restarted the comparison test with some extra checks to make sure that if comparison results are not uploaded then do not update "comparison is Ready" message and retry..

**Comment by mmusich** [comment]
+1

@civanch, @dmitrijus, @cvuosalo, @franzoni, @mdhildreth, @fabozzi, @slava77, @vanbesien, @hengne, @davidlange6 

could you please review and eventually approve? 
We're going to need the cosmics workflow for alignment purposes in 810.

**Comment by boudoul** [comment]
as TK-DPG I second the request from @mmusich , thanks!

**Comment by fabozzi** [comment]
+1

**Comment by dmitrijus** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by cvuosalo** [comment]
@mmusich, @schneiml: Are you saying that you enabled DQM plots for the new 7.3 workflow (https://github.com/cms-sw/cmssw/pull/16343#issuecomment-258490047)? I tested the latest version of this PR using CMSSW_8_1_X_2016-11-04-1100 and found that workflow 7.3 still has 12510 missing DQM plots compared with the very similar workflow 1307.0_CosmicsSPLoose_UP15.

**Comment by schneiml** [comment]
@cvuosalo I only turned on the new Pixel DQM, but I'd guess nothing changed for the Strips, so it should run as well.

Comparing the number of plots (or even checking for MEs not present that were there before) is for sure the wrong methodology, since the Phase1 Pixel DQM is completely rebuilt from scratch, and *all* Pixel names changed (if only due to the new top folder name). It happens that 12510 sounds a bit like the total number of Phase0 Pixel plots (+-10k) so I'd say this is expected.

OTOH, you *should* see a `SiStrip/` and `PixelPhase1/` folder in the DQM output, otherwise something is broken.

**Comment by cvuosalo** [comment]
With the latest version of the PR, I repeated the comparison of the new 7.3 workflow with 1307.0_CosmicsSPLoose_UP15 using 1000 events. Here are the new versions of the plots that showed problems previously (https://github.com/cms-sw/cmssw/pull/16343#issuecomment-257983051). The second plot below, at least, looks suspicious. "New" is 7.3, and "Ref" is 1307.0.

all_testpr16343vsorig_cosmicsup15wf1307p0c_calotowerssorted_towermaker__reco_obj_obj_eta123
all_testpr16343vsorig_cosmicsup15wf1307p0c_hfrechitssorted_hfreco__reco_obj_obj_energy

all_testpr16343vsorig_cosmicsup15wf1307p0c_log10horechitssorted_horeco__reco_obj_obj_energy3

all_testpr16343vsorig_cosmicsup15wf1307p0c_log10hbherechitssorted_hbhereco__reco_obj_obj_energy1

**Comment by cvuosalo** [comment]
@schneiml: The new 7.3 workflow DQM is still incomplete. The missing plots include CSC, RPC, and SiStrip plots. This new workflow needs to cover all the standard DQM plots.

**Comment by kpedro88** [comment]
@igv4321 can you double-check the HF reco settings/behavior for cosmics?

@cvuosalo, myself and @abdoulline are working on debugging a potential issue with HO and HB conditions for 2017 that might be causing problems both here and in normal 2017 workflows.

**Comment by kpedro88** [comment]
@cvuosalo would prefer not to hold up this PR for those issues... they will be addressed eventually.

**Comment by igv4321** [comment]
Yes, the settings look good to me. The energy distribution does
look funny. My guess is that the pedestals are not yet realistic and
do not result in the smearing of the rechit energies.

**Comment by kpedro88** [comment]
The pedestals were set according to the QIE10 calibration information I was given in the spring, and should be the same for all 2017 tags/workflows... but we are currently using the same value for all channels (13.33 fC +/- 3.33 fC). Maybe we would see this in normal workflows also if we looked at a very low energy range.

**Comment by cvuosalo** [comment]
@kpedro88: I think it would probably be acceptable to address the discrepancies in the distributions in a later PR, but the thousands of missing DQM plots need to be fixed to make the 7.3 workflow properly functional.

**Comment by kpedro88** [comment]
Just checking, none of the missing plots are from HCAL, right? Our validation/DQM has been running for 2017 for a while and shouldn't have any problems with the cosmic scenario.

**Comment by cvuosalo** [comment]
@kpedro88: I don't see any missing HCAL plots, and I actually found a number of newly filled-in DQM HCAL plots. I should correct the number of missing plots I mentioned above. There are probably thousands of missing DQM plots, of the types I mention above, but they don't total 12510. Some of the 12510 are correct plots that are newly filled in or just have minor differences.

**Comment by mmusich** [comment]
@cvuosalo can you actually supply few examples of the missing histograms? May I ask if @dmitrijus sees something suspicious in the cmsDriver configurations for the newly created 7.3 wf?

**Comment by cvuosalo** [comment]
@mmusich: I will list several categories of missing plots, along with a few specific examples. Note that there may be other categories of missing plots in addition to the ones I've listed.

Category: /DQMData/Run 1/CSC/Run summary/CSCOfflineMonitor/
For example:/DQMData/Run 1/CSC/Run summary/CSCOfflineMonitor/Segments, Segment chi2 Probability

Category: /DQMData/Run 1/DT/Run summary/02-Segments
For example: /DQMData/Run 1/DT/Run summary/02-Segments/Wheel-1/Sector4/Station3, # of hits per segment

Category: /DQMData/Run 1/EcalBarrel/Run summary
For example: /DQMData/Run 1/EcalBarrel/Run summary/EBTimingTask, EBTMT EB Chi2

Category: /DQMData/Run 1/RPC/Run summary/AllHits/Barrel
For example: /DQMData/Run 1/RPC/Run summary/AllHits/Barrel/Wheel_-1/sector_12/station_4, ClusterSize_W-1_RB4-_S12

**Comment by mmusich** [comment]
for the record:

```
7.3    CosmicsSPLoose_UP17+DIGICOS_UP17+RECOCOS_UP17+ALCACOS_UP17+HARVESTCOS_UP17 [1]: cmsDriver.py UndergroundCosmicSPLooseMu_cfi.py  --conditions auto:phase1_2017_cosmics -s GEN,SIM -n 10 --era Run2_2017 --eventcontent FEVTDEBUG --relval 5000000,500000 --scenario cosmics --datatier GEN-SIM --beamspot Realistic50ns13TeVCollision 
                                           [2]: cmsDriver.py step2  --conditions auto:phase1_2017_cosmics -s DIGI:pdigi_valid,L1,DIGI2RAW,HLT:@fake --scenario cosmics --datatier GEN-SIM-DIGI-RAW -n 10 --era Run2_2017 --eventcontent FEVTDEBUG
                                           [3]: cmsDriver.py step3  --conditions auto:phase1_2017_cosmics -s RAW2DIGI,L1Reco,RECO,ALCA:MuAlGlobalCosmics,DQM --scenario cosmics --datatier GEN-SIM-RECO,DQMIO -n 10 --era Run2_2017 --eventcontent RECOSIM,DQM
                                           [4]: cmsDriver.py step4  --conditions auto:phase1_2017_cosmics -s ALCA:TkAlCosmics0T+TkAlBeamHalo+MuAlBeamHaloOverlaps+MuAlBeamHalo --datatier ALCARECO -n 1000 --era Run2_2017 --eventcontent ALCARECO
                                           [5]: cmsDriver.py step5  --conditions auto:phase1_2017_cosmics --filein file:step3_inDQM.root -s HARVESTING:dqmHarvesting --filetype DQM --scenario cosmics --era Run2_2017 --mc  -n 100 
```

```
1307.0 CosmicsSPLoose_UP15+DIGICOS_UP15+RECOCOS_UP15+ALCACOS_UP15+HARVESTCOS_UP15 [1]: cmsDriver.py UndergroundCosmicSPLooseMu_cfi.py  --conditions auto:run2_mc_cosmics -s GEN,SIM -n 10 --era Run2_2016 --eventcontent FEVTDEBUG --relval 5000000,500000 --scenario cosmics --datatier GEN-SIM --beamspot Realistic50ns13TeVCollision 
                                           [2]: cmsDriver.py step2  --conditions auto:run2_mc_cosmics -s DIGI:pdigi_valid,L1,DIGI2RAW,HLT:@relval2016 --scenario cosmics --datatier GEN-SIM-DIGI-RAW -n 10 --era Run2_2016 --eventcontent FEVTDEBUG
                                           [3]: cmsDriver.py step3  --conditions auto:run2_mc_cosmics -s RAW2DIGI,L1Reco,RECO,ALCA:MuAlGlobalCosmics,DQM --scenario cosmics --datatier GEN-SIM-RECO,DQMIO -n 10 --era Run2_2016 --eventcontent RECOSIM,DQM
                                           [4]: cmsDriver.py step4  --conditions auto:run2_mc_cosmics -s ALCA:TkAlBeamHalo+MuAlBeamHaloOverlaps+MuAlBeamHalo --datatier ALCARECO -n 1000 --era Run2_2016 --eventcontent ALCARECO
                                           [5]: cmsDriver.py step5  --conditions auto:run2_mc_cosmics --filein file:step3_inDQM.root -s HARVESTING:dqmHarvesting --filetype DQM --scenario cosmics --era Run2_2016 --mc  -n 100 
```

**Comment by mmusich** [comment]
@dmitrijus excepted the conditions which are different by construction, the only difference I see is that 7.1 is running `HLT:@fake` while 1307.0 is running `HLT@relval2016`
might there be a specific HLT paths selection on the plot categories listed by https://github.com/cms-sw/cmssw/pull/16343#issuecomment-259375996 ?

**Comment by mmusich** [comment]
please test

**Comment by mmusich** [comment]
@cvuosalo with the last commit I can get non-empty histograms for example for:

Category: /DQMData/Run 1/DT/Run summary/02-Segments
For example: /DQMData/Run 1/DT/Run summary/02-Segments/Wheel-1/Sector4/Station3, # of hits per segment

would you please cross-check? thanks

**Comment by slava77** [comment]
On 11/9/16 1:11 PM, Marco Musich wrote:
> @cvuosalo <https://github.com/cvuosalo> with the last commit I can get
> non-empty histograms for example for:
> 
> Category: /DQMData/Run 1/DT/Run summary/02-Segments
> For example: /DQMData/Run 1/DT/Run
> summary/02-Segments/Wheel-1/Sector4/Station3, # of hits per segment
> 
> would you please cross-check? thanks
> 

I see that only the HLT part of the setup has changed.
Does it mean there are trigger selections in the muon DQM plots?
... or does it mean that these modules somehow rely on muon reco
products that are recovered by HLT
and are missing from the reco step.

**Comment by mmusich** [comment]
@slava77 I don't know, see my comment: https://github.com/cms-sw/cmssw/pull/16343#issuecomment-259376701. 
At least I am quite certain that Strip DQM is filtered on specific HLT paths, I don't know for Muons.

**Comment by hengne** [comment]
@mmusich Hi Marco,  this should be implemented in the upgrade framework:  upgradeWorkflowComponents.py

**Comment by mmusich** [comment]
@hengne see https://github.com/cms-sw/cmssw/pull/16343#discussion_r84902398

**Comment by cvuosalo** [comment]
@mmusich: When I run the latest version of workflow 7.3 in CMSSW_8_1_X_2016-11-04-1100, I get the following empty plot, compared with 1307.0.
dt

**Comment by mmusich** [comment]
@cvuosalo I don't know how you proceed to produce those plots, but if I follow this recipe:

```
cmsrel CMSSW_8_1_X_2016-11-10-1100
cd CMSSW_8_1_X_2016-11-10-1100/src/
git cms-init
git remote add cms-AlCaDB git@github.com:cms-AlCaDB/cmssw.git
git fetch cms-AlCaDB
git cms-addpkg Configuration/AlCa
git cms-addpkg Configuration/PyReleaseValidation
git cms-addpkg RecoLocalCalo/Configuration
git cms-addpkg DQM/SiPixelCommon
git cms-addpkg DQM/SiPixelPhase1Config
git cms-addpkg DQMOffline/Configuration
git cms-addpkg SimCalorimetry/HcalSimProducers
git cms-addpkg SimCalorimetry/HcalZeroSuppressionProducers
git pull cms-AlCaDB PhaseI_cosmics
scramv1 b -j 8
cmsenv
runTheMatrix.py -l 7.3 --command='-n 10000'
```

I end up with this:

screenshot 2016-11-10 14 57 23

puzzled...

**Comment by cvuosalo** [comment]
@mmusich: It looks like there is a 1000 times difference in efficiency. My test of workflow 1307.0 for 1000 events produced 206 entries in the plot (https://github.com/cms-sw/cmssw/pull/16343#issuecomment-259455351), so efficiency is 0.206, while your test used 10000 events and produced 3 entries, for an efficiency of 0.003. Do you know why the new workflow 7.3 would have so few entries for this plot compared to Run 2 cosmics workflow 1307.0?

I ran my test with:

cmsrel  CMSSW_8_1_X_2016-11-04-1100
...
git cms-merge-topic --unsafe 16343
git cms-checkdeps -a
...
runTheMatrix.py -l 7.3 --useInput all --command="-n 1000"

**Comment by mmusich** [comment]
@cvuosalo in the same installation prepared as before running 
```
runTheMatrix.py -l 1307.0 --command='-n 10000'
```

I get this: 

screenshot 2016-11-10 19 05 28

But the way I do get the very same when running:
```
cmsrel CMSSW_8_1_X_2016-11-04-1100
cd CMSSW_8_1_X_2016-11-04-1100/ CMSSW_8_1_X_2016-11-10-1100/ 
cmsenv
runTheMatrix.py -l 1307.0 --command='-n 10000'
```

What does `--useInput all` do? recycles GEN-SIM? This wf has a very low generator level efficiency skimming events in the pixel volume acceptance, if you don't re-do the GEN-SIM from scratch you are making an unfair comparison.

**Comment by mmusich** [comment]
@kpedro88 @abdoulline are the conditions proposed in https://github.com/cms-sw/cmssw/pull/16544 supposed to fix the issues reported in https://github.com/cms-sw/cmssw/pull/16343#issuecomment-259187622 ?

**Comment by kpedro88** [comment]
@mmusich the new conditions should fix HB and HO

**Comment by mmusich** [comment]
last update picks up same Hcal conditions as in https://github.com/cms-sw/cmssw/pull/16544
```
$ conddb diff 81X_upgrade2017cosmics_realistic_peak_v5 81X_upgrade2017cosmics_realistic_peak_v6
[2016-11-10 21:14:48,969] INFO: Connecting to pro [frontier://PromptProd/CMS_CONDITIONS]
Record                      Label  pro::81X_upgrade2017cosmics_realistic_peak_v5 Tag  pro::81X_upgrade2017cosmics_realistic_peak_v6 Tag  
--------------------------  -----  -------------------------------------------------  -------------------------------------------------  
HcalGainsRcd                -      HcalGains_2017_v1.0_mc                             HcalGains_2017_v2.0_mc                             
HcalPedestalWidthsRcd       -      HcalPedestalWidths_2017_v2.0_mc                    HcalPedestalWidths_2017_v3.0_mc                    
HcalPedestalsRcd            -      HcalPedestals_2017_v2.0_mc                         HcalPedestals_2017_v3.0_mc                         
HcalSiPMCharacteristicsRcd  -      HcalSiPMCharacteristics_2017_v5.0_mc               HcalSiPMCharacteristics_2017_v6.0_mc               
HcalSiPMParametersRcd       -      HcalSiPMParameters_2017_v4.0_mc                    HcalSiPMParameters_2017_v5.0_mc                    
HcalTPParametersRcd         -      HcalTPParameters_2017_v1.0_mc                      HcalTPParameters_2017_v2.0_mc  
```

**Comment by mmusich** [comment]
please test

**Comment by mmusich** [comment]
@cmsbuild, please test
needed to rebase after mergin of #16544

**Comment by cvuosalo** [comment]
@mmusich: Re-base still needed.

**Comment by kpedro88** [comment]
I think that's now because of #16461, @mmusich let me know if the rebase isn't straightforward

**Comment by mmusich** [comment]
I am on leave with limited access to stable connection until tuesday. Will rebase once back. @cvuosalo I'd appreciate speed up in review and approval if there are no further comments to be adressed

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
@cmsbuild please test
@kpedro88 indeed the last changes #16461 broke this one. Can you check if I did the correct thing here: https://github.com/cms-sw/cmssw/pull/16343/commits/eebeb4430bfaa8e06ec99d1dfa75d9f9bff9bcbf

**Comment by kpedro88** [comment]
@mmusich that should be correct

**Comment by kpedro88** [comment]
```
config/SCRAM/GMake/Makefile.rules:1965: recipe for target 'unittests_TestIOMCRandomEngineService' failed
gmake: *** [unittests_TestIOMCRandomEngineService] Terminated
```
I'm not sure what that means...

**Comment by davidlange6** [comment]
please test

seems a glitch

**Comment by cvuosalo** [comment]
I tested with the version of this PR just prior to the last re-base. I see the same empty DQM plots as before (https://github.com/cms-sw/cmssw/pull/16343#issuecomment-259455351), and similar discrepancies in the validation plots (https://github.com/cms-sw/cmssw/pull/16343#issuecomment-259187622). What is the plan for fixing these problems?

**Comment by kpedro88** [comment]
@cvuosalo the HO and HB discrepancies should be resolved with the latest GT conditions (not sure what you used in your test, or if @mmusich has updated the affected records in the cosmic GT). There will of course be a change in HE response. The HF change may just be due to using the same pedestal for each channel (combined with Poisson behavior of the HF simulation, which delivers SimHits in units of photoelectrons instead of sensitive layer energy deposits). The CaloTower thresholds are known to need retuning and this is JetMET's responsibility.

In summary, none of those discrepancies are unique to the cosmic workflow, some have been understood, and others are either expected or being investigated.

**Comment by mmusich** [comment]
please test

**Comment by mmusich** [comment]
@cvuosalo about:

> I see the same empty DQM plots as before (#16343 (comment)), 

I am wondering if my last comment about it escaped your attention.
I am reposting it here for your convenience:

======
======
in the same installation prepared as before running 
```
runTheMatrix.py -l 1307.0 --command='-n 10000'
```

I get this: 

screenshot 2016-11-10 19 05 28

But the way I do get the very same when running:
```
cmsrel CMSSW_8_1_X_2016-11-04-1100
cd CMSSW_8_1_X_2016-11-04-1100/ CMSSW_8_1_X_2016-11-10-1100/ 
cmsenv
runTheMatrix.py -l 1307.0 --command='-n 10000'
```

What does `--useInput all` do? recycles GEN-SIM? 
This wf has a very low generator level efficiency skimming events in the pixel volume acceptance, if you don't re-do the GEN-SIM from scratch you are making an unfair comparison, because the new wf 7.3 has no GEN-SIM to be recycled.

**Comment by davidlange6** [comment]
please test
(seemed a glitch)

**Comment by cvuosalo** [comment]
@mmusich: Yes, I did the GENSIM step in my tests. I ran 10,000 events and have seen fewer DQM discrepancies, so it may be a statistics issue. I will run 900,000 events and check the DQM plots again.

**Comment by mmusich** [comment]
please test

**Comment by mmusich** [comment]
+1

**Comment by mmusich** [comment]
@cvuosalo can we try to converge before I need to rebase the 10th time? thanks

**Comment by cvuosalo** [comment]
@mmusich: There are now new differences in the Jenkins test results that didn't appear in the earliest Jenkins tests. Should there be HCAL differences in Phase 1 and Phase 2 workflows like the ones below? (Red is the PR, black is the baseline.)

tim2
timing
fed

**Comment by kpedro88** [comment]
@cvuosalo at some point recently we updated a bunch of HCAL parameters. I can't keep track of what GT was used for what test at this point. Let's just move forward with the PR.

**Comment by kpedro88** [comment]
@cvuosalo after more inspection, I think all the changes come from changing the HF QIE10 digi collection to store only 4 samples (as is done in the data) instead of 10. Some of the DQM plots assumed that all 10TS should be used in calculating quantities.

**Comment by cvuosalo** [comment]
+1

For #16343 1c5bc2b6585410ff5522bd6b828544481a5ba2c7

Adding a new Phase 1 cosmics workflow.

#16709 is the 90X version of this PR.

The code changes are satisfactory. Jenkins tests against baseline CMSSW_8_1_X_2016-11-21-1100 show some small differences that are expected from this PR, as discussed above. A test of the new workflow compared with workflow 1307.0_CosmicsSPLoose_UP15 for 900,000 events indicates the new workflow has basic functionality, though there are some small discrepancies that will be resolved later, as discussed above (https://github.com/cms-sw/cmssw/pull/16343#issuecomment-261105519).
